metadata_backend/helpers/metax_api_handler.py

```diff
@@ -0,0 +1,261 @@
+        :returns: Metax ID for dataset returned by Metax API
+        """
+        metax_dataset = self.minimal_dataset_template
+        metax_dataset["metadata_provider_user"] = await self.get_metadata_provider_user()
+        if collection == "dataset":
+            dataset_data = await self.create_metax_dataset_data_from_dataset(data)
+        else:
+            dataset_data = await self.create_metax_dataset_data_from_study(data)
+        metax_dataset["research_dataset"] = dataset_data
+        LOG.debug(
+            f"Creating draft dataset to Metax service from Submitter {collection} with accession ID "
+            f"{data['accessionId']}"
+        )
+        async with aiohttp.ClientSession() as sess:
+            resp = await sess.post(
+                f"{self.metax_url}{self.rest_route}",
+                params="draft",
+                json=metax_dataset,
+                auth=aiohttp.BasicAuth(self.username, self.password),
+            )
+            status = resp.status
+            if status == 201:
+                metax_data = await resp.json()
+                LOG.debug(
+                    f"Created Metax draft dataset {metax_data['identifier']} from Submitter {collection} "
+                    f"{data['accessionId']} with data: {metax_dataset}."
+                )
+                return metax_data["identifier"]
+            else:
+                # TODO: how front end should react on this??
+                reason = await resp.text()
+                raise self.process_error(status, reason)
+
+    async def update_draft_dataset(self, collection: str, data: Dict) -> str:
+        """Update draft dataset to Metax.
+
+        Construct Metax draft dataset data from submitters' Study or Dataset and
+        send it to Metax Dataset API for update.
+
+        :param collection: schema of incomming submitters metadata
+        :param data: validated Study or Dataset data dict
+        :raises: HTTPError depending on returned error from Metax
+        :returns: Metax ID for dataset returned by Metax API
+        """
+        metax_dataset = self.minimal_dataset_template
+        # TODO: should this be changed if person updating data is different from data creator?
+        metax_dataset["metadata_provider_user"] = await self.get_metadata_provider_user()
+        if collection == "dataset":
+            dataset_data = await self.create_metax_dataset_data_from_dataset(data)
+        else:
+            dataset_data = await self.create_metax_dataset_data_from_study(data)
+        metax_dataset["research_dataset"] = dataset_data
+        LOG.info(f"Sending updated {collection} object data to Metax service.")
+
+        async with aiohttp.ClientSession() as sess:
+            resp = await sess.put(
+                f'{self.metax_url}{self.rest_route}/{data["metaxIdentifier"]}',
+                params="draft",
+                json=metax_dataset,
+                auth=aiohttp.BasicAuth(self.username, self.password),
```

Comment:
can users see our service account credentials in browser network-tab?

Reply:
I don't think that is the case as the client will be the backend app in this case not the browser so the connection is initiated in the backend. I tried it and it does not seem to be the case.

However if you notice it let us know, this affects DOIHandler as well.